sensor/kubernetes/eventpipeline/pipeline_impl.go

```diff
@@ -93,7 +94,25 @@ func (p *eventPipeline) Stop(_ error) {
 	p.stopSig.Signal()
 }
 
-func (p *eventPipeline) Notify(common.SensorComponentEvent) {}
+func (p *eventPipeline) Notify(event common.SensorComponentEvent) {
+	log.Infof("Received notify: %s", event)
+	switch event {
+	case common.SensorComponentEventCentralReachable:
+
+		// Start listening to events if not yet listening
+		if p.offlineMode.CompareAndSwap(false, true) {
+			log.Infof("Connection established: Starting Kubernetes listener")
+			if err := p.listener.Start(); err != nil {
+				log.Fatalf("Failed to start listener component. Sensor cannot run without listening to Kubernetes events: %s", err)
+			}
+		}
+	case common.SensorComponentEventOfflineMode:
+		// Stop listening to events
+		if p.offlineMode.CompareAndSwap(true, false) {
```

Comment:
Yea you're right, seems like the variable is being used as a `isListeningToEvents` flag instead, that's why it works.  

I could rename it, but perhaps it's better to keep it as `offlineMode` since in v2 we'll probably want to introduce other behaviors attached to being offline rather than listening or not. I'll make the changes.